lib/src/gc.rs

```diff
@@ -98,106 +91,129 @@ impl<'r, I, J> CollectableRefs<'r, I, J>
 
-                // Whether the local head should be collected or not is computed
-                // here, in the exact same way it is for leaves. We do that
-                // because can't mix the computation with those of the leaves.
-                // It would cause head references to be removed if any message
-                // was posted as a reply to the current head.
-                let mut head_history = self
-                    .repo
-                    .revwalk()
-                    .chain_err(|| EK::CannotConstructRevwalk)?;
-                match self.collect_heads {
-                    ReferenceCollectionSpec::Never => {},
-                    ReferenceCollectionSpec::BackedByRemoteHead => {
-                        for item in issue.remote_refs(IssueRefType::Head)? {
-                            head_history.push(
-                                item?
-                                    .peel(git2::ObjectType::Commit)
-                                    .chain_err(|| EK::CannotGetCommit)?
-                                    .id()
-                            )?;
-                        }
-                    },
-                };
-                let mut referring_refs = iter::RefsReferringTo::new(head_history);
-                referring_refs.watch_ref(local_head)?;
-                referring_refs.collect_result_into(&mut retval)?;
+        // remote refs
+        if self.consider_remote_refs {
+            for item in issue.remote_refs(IssueRefType::Any)? {
+                retval.push(item?
+                    .peel(git2::ObjectType::Commit)
+                    .chain_err(|| EK::CannotGetCommit)?
+                    .id()
+                )?;
             }
+        }
 
-            // local leaves
-            for item in issue.local_refs(IssueRefType::Leaf)? {
-                let leaf = item?;
-                // NOTE: We push the parents of the references rather than the
-                //       references themselves since that would cause the
-                //       `RefsReferringTo` report that exact same reference.
-                Self::push_ref_parents(&mut messages, &leaf)?;
-                refs_to_assess.push(leaf);
-            }
+        Ok(retval)
+    }
 
-            // remote refs
-            if self.consider_remote_refs {
-                for item in issue.remote_refs(IssueRefType::Any)? {
-                    messages.push(item?
-                        .peel(git2::ObjectType::Commit)
-                        .chain_err(|| EK::CannotGetCommit)?
-                        .id()
-                    )?;
-                }
-            }
-        }
+    /// Find collectable references for multiple issues
+    ///
+    /// This is a convenience function.
+    ///
+    /// # Note
+    ///
+    /// Internally, this function collects the references during the call.
+    /// Consider using the `for_issues` function in conjunction with
+    /// `flat_map()` on an issue iterator instead.
```

Comment:
Why making this function `pub` if you discourage users here? Why not simply make it private?

Reply:
It was `pub` previously. Effectively, it's a legacy API which will vanish at some point.

Reply:
We could mark it as deprecated then: https://stackoverflow.com/questions/30757914/can-i-mark-a-function-as-deprecated